unstructured/chunking/title.py

```diff
@@ -9,105 +10,153 @@
+                table_chunk.metadata.text_as_html = text_as_html_chunk
+                # Remove the processed chunk from text_as_html
+                text_as_html = text_as_html[max_characters:]
+            if i > 0:
+                table_chunk.metadata.is_continuation = True
+
+            chunks.append(table_chunk)
+            i += 1
+
+            # Remove the processed chunk from text
+            text = text[max_characters:]
+
+            # Ensure that text and text_as_html are not empty before continuing
+            if not text and not text_as_html:
+                break
+    return chunks
+
+
 def chunk_by_title(
     elements: List[Element],
     multipage_sections: bool = True,
-    combine_under_n_chars: int = 500,
-    new_after_n_chars: int = 1500,
+    combine_text_under_n_chars: int = 500,
+    max_characters: int = 1500,
 ) -> List[Element]:
     """Uses title elements to identify sections within the document for chunking. Splits
     off into a new section when a title is detected or if metadata changes, which happens
     when page numbers or sections change. Cuts off sections once they have exceeded
-    a character length of new_after_n_chars.
+    a character length of max_characters.
 
     Parameters
     ----------
     elements
-        A list of unstructured elements. Usually the ouput of a partition functions.
+        A list of unstructured elements. Usually the output of a partition functions.
     multipage_sections
         If True, sections can span multiple pages. Defaults to True.
-    combine_under_n_chars
+    combine_text_under_n_chars
         Combines elements (for example a series of titles) until a section reaches
         a length of n characters.
-    new_after_n_chars
+    max_characters
         Cuts off new sections once they reach a length of n characters
     """
     if (
-        combine_under_n_chars is not None
-        and new_after_n_chars is not None
+        combine_text_under_n_chars is not None
+        and max_characters is not None
         and (
-            combine_under_n_chars > new_after_n_chars
-            or combine_under_n_chars < 0
-            or new_after_n_chars < 0
+            combine_text_under_n_chars > max_characters
+            or combine_text_under_n_chars < 0
+            or max_characters < 0
         )
     ):
         raise ValueError(
-            "Invalid values for combine_under_n_chars and/or new_after_n_chars.",
+            "Invalid values for combine_text_under_n_chars and/or max_characters.",
         )
 
     chunked_elements: List[Element] = []
     sections = _split_elements_by_title_and_table(
         elements,
         multipage_sections=multipage_sections,
-        combine_under_n_chars=combine_under_n_chars,
-        new_after_n_chars=new_after_n_chars,
+        combine_text_under_n_chars=combine_text_under_n_chars,
+        max_characters=max_characters,
     )
-
     for section in sections:
```

Comment:
This may be a good opportunity to clean up this loop a bit as it's getting unwieldy
```
    for section in sections:
        if not section:
            continue

        first_element = section[0]
        
        if not isinstance(first_element, Text):
            chunked_elements.extend(section)
            continue
        
        if isinstance(first_element, Table):
            chunked_elements.extend(chunk_table_element(first_element, max_characters))
            continue

        text = ""
        metadata = first_element.metadata
        start_char = 0

        for element in section:
            if isinstance(element, Text):
                text += "\n\n" if text else ""
                start_char = len(text)
                text += element.text

            for attr, value in vars(element.metadata).items():
                if isinstance(value, list):
                    _value = getattr(metadata, attr, []) or []
                    
                    if attr == "regex_metadata":
                        for item in value:
                            item["start"] += start_char
                            item["end"] += start_char

                    _value.extend(item for item in value if item not in _value)
                    setattr(metadata, attr, _value)

        chunked_elements.append(CompositeElement(text=text, metadata=metadata))

    return chunked_elements
```

Reply:
oof, agree; simple step might even be taking the element loop out into own function first so that:
- can do unit test
- reduce nesting
now more to this might involve data structure; we store data as intuitive objects right now but may not be most efficient for compute; one alternative is to store data as block (e.g., numpy array) and attributes are just reference to the data underneath. This makes expression still natural but computation can be better optimized. But that's really for another ticket on its own; for now lets at least refactor the inner loop out

Reply:
+1

Reply:
Sounds good! Can definitely simplify the existing loop, and I'll make note of this convo for the chunking work for other element types that is in the next sprint :)